### PR TITLE
"examples" implementation for material-ui

### PR DESCRIPTION
**Pull request by inrumi**
### Reasons for making this change

Related to the issue I raised heh 👉👈  #2368 

This wasn't really complicated, most of the heavy lifting was in the bootstrap implementation.

If test are required I can work on them this weekend.

### Checklist

* [ ] **I'm updating documentation**
  - [ ] I've [checked the rendering](https://react-jsonschema-form.readthedocs.io/en/latest/#contributing) of the Markdown text I've added
* [x] **I'm adding or updating code**
  - [x] I've added and/or updated tests
  - [ ] I've updated [docs](https://react-jsonschema-form.readthedocs.io/) if needed
* [ ] **I'm adding a new feature**
  - [ ] I've updated the playground with an example use of the feature

### Deploy preview
image

Once CI finishes, you should be able to view a deploy preview of the playground from this PR at this link: https://deploy-preview-[PR_NUMBER]--rjsf.netlify.app/

### CODE REVIEW

**Comment by inrumi** [comment]
@epicfaace is there any chance I can get a review here 🙏?

**Review by epicfaace — COMMENTED** [review]
Thanks for the PR -- it would be great to add tests (snapshot tests) if you're able.

**Comment by inrumi** [comment]
> Thanks for the PR -- it would be great to add tests (snapshot tests) if you're able.

Done! :D

**Comment by inrumi** [comment]
@epicfaace hi again! so I added the tests, everything is ok for this PR to be reviewed/merged?

**Review by epicfaace — APPROVED** [review]
Looks good, thanks!

File: packages/material-ui/src/TextWidget/TextWidget.tsx [hunk 1]
```diff
@@ -46,24 +47,39 @@ const TextWidget = ({
...
+        required={required}
+        disabled={disabled || readonly}
+        type={inputType as string}
+        value={value || value === 0 ? value : ""}
+        error={rawErrors.length > 0}
+        onChange={_onChange}
+        onBlur={_onBlur}
+        onFocus={_onFocus}
+        inputProps={{
+          list: `examples_${id}`,
```
Comment (epicfaace) [inline packages/material-ui/src/TextWidget/TextWidget.tsx:69]: We shouldn't include this prop if `schema.examples` doesn't exist.

**Review by epicfaace — CHANGES_REQUESTED** [review]
Can you also update snapshots by running `npm test -- -u`?

**Comment by Jame-Stitel** [comment]
Please, we need this implementation... What needs to be done?

**Comment by heath-freenome** [comment]
Fixed in v5 beta via refactor of this PR

**Pull request closed without merging**